test.js

```diff
@@ -166,6 +166,20 @@ if (isNode8orHigher) {
 	});
 }
 
+test('is.generator', t => {
+	const genObj = (function * () {
+		yield 42;
+	})();
+	t.true(m.generator(genObj));
+});
```

Comment:
Use the `testType` type of testing, so it tests against all the other tests too. You probably need to exclude `function`.

Reply:
Hey @sindresorhus 👋

Stuck here. I was going through your `types` map, and was hoping to just plug in a generator function that returns an object, but that doesn't seem to suffice. Am I missing something here?

For reference:

```diff
+ ['generator', (function* () { yield 42; })()],
```

... and calling it as so:

```js
test('is.generator', t => {
	testType(t, 'generator', ['function']);
});
```

... gives me:

```
Value: undefined

  Value is not `false`:

  undefined
```

... although I believe, it should return a generator object, and hence, pass the test. :thinking:

Reply:
It's because of the `x &&` part of the detection. If it's false, it returns itself instead of false, so `undefined` would return `undefined` instead of `false.`

Reply:
Ah, I see `undefined &&` will always return `undefined`.

However, it's weird because in all my testing so far, calling a generator function returns a generator object, so I'm not sure where it's returning `undefined`, and therefore breaking the chain.

I'll look further into it tho. Thanks for the input so far. Appreciate it. :')